inc/Menu/Item/Revert.php

```diff
@@ -7,20 +7,29 @@
  *
  * Quick revert to the currently shown page revision
  */
-class Revert extends AbstractItem {
+class Revert extends AbstractItem
+{
 
     /** @inheritdoc */
-    public function __construct() {
+    public function __construct()
+    {
         global $REV;
         global $INFO;
+        global $INPUT;
         parent::__construct();
 
-        if(!$INFO['ismanager'] || !$REV || !$INFO['writable']) {
+        // if we are comparing with the current version then
+        // allow reverting to the non current version
+        $comparedWithCurrent = in_array('current', $INPUT->ref('rev2')) || in_array('', $INPUT->ref('rev2'));
```

Comment:
`rev2` may not be set at all (for basically all page views except for when creating a diff) so this will throw a warning. I wonder if `ref` should be `array` instead?

Reply:
Can you post the files/errors? I pulled upstream changes and am not seeing them.

Reply:
I have these:

```
Warning: in_array() expects parameter 2 to be array, string given in ...\inc\Menu\Item\Revert.php on line 23

Warning: in_array() expects parameter 2 to be array, string given in ...\inc\Menu\Item\Revert.php on line 23
```

To enable `E_ALL` errors globally, create the file 'conf/report_e_all'.

Reply:
@kalenpw what's the status here? Can you fix the remaining issues?

Reply:
I believe so as they seem relatively minor, but I am still not able to reproduce the errors. I created `conf/report_e_all`, set `error_reporting = E_ALL` in `php.ini`, and added `error_reporting(E_ALL);` to `doku.php` for good measure. Is there something obvious I am missing?

Reply:
I've also noticed now it works when you first select current revision + old revision, but if you then change to current revision + any other old revision you can no longer restore. 

Does that have to do with any of the changes to `$INPUT->ref('rev2')`?